packages/matter.js/src/util/Promises.ts

```diff
@@ -34,3 +34,30 @@ export function createPromise<T>(): {
         rejecter,
     };
 }
+
+/**
+ * Use all promises or promise returning methods and return the first resolved promise or reject when all promises
+ * rejected
+ */
+export function anyPromise<T>(promises: ((() => Promise<T>) | Promise<T>)[]): Promise<T> {
```

Comment:
This is nice but I really feel like we should just use Promise.any (node 15+, deno/browsers since 2020).  That would give us AggregateError too which would make the rejection error more accurate.  We can just bump lib in tsconfig.base.json to es2021 and stay with the older target until that gets unreasonably annoying too. ;)

Reply:
I would like to do that in a separate PR then ... ok? Added todo https://github.com/orgs/project-chip/projects/11/views/1?pane=issue&itemId=43777847

Reply:
Sounds good